test/Unicode/CharSpec.hs

```diff
@@ -0,0 +1,81 @@
+{-# LANGUAGE BlockArguments #-}
+{-# LANGUAGE CPP #-}
+
+module Unicode.CharSpec
+  ( spec
+  ) where
+
+import qualified Data.Char as Char
+import qualified Unicode.Char as UChar
+import Data.Foldable (traverse_)
+import Test.Hspec
+
+{- [NOTE] 
+These tests may fail if the compiler’s Unicode version
+does not match the version of this package.
+
++-------------+----------------+-----------------+
+| GHC version | @base@ version | Unicode version |
++=============+================+=================+
+| 8.8         | 4.13           | 12.0            |
+| 8.10.[1-4]  | 4.14.{0,1}     | 12.0            |
+| 8.10.5+     | 4.14.2+        | 13.0            |
+| 9.0.1       | 4.15.0         | 12.1            |
+| 9.2.1       | 4.16.0         | 14.0            |
++-------------+----------------+-----------------+
+-}
+
+spec :: Spec
+spec = do
+#ifdef COMPATIBLE_GHC_UNICODE
+  let describe' = describe
+#else
+  let describe' t = before_ (pendingWith "Incompatible GHC Unicode version") . describe t
+#endif
+  describe' "Unicode general categories" do
+    it "generalCategory" do
+      -- Note: we cannot compare the categories directly, so use 'show'.
+      let sameGC c = show (UChar.generalCategory c) == show (Char.generalCategory c)
+      traverse_ (`shouldSatisfy` sameGC) [minBound..maxBound]
+  describe' "Character classification" do
+    it "isAlpha" do
+      let same c = UChar.isAlpha c == Char.isAlpha c
+      traverse_ (`shouldSatisfy` same) [minBound..maxBound]
```

Comment:
This pattern is repeating in all tests, can be factored out as a function.

Reply:
Fixed